pkg/operator/controller/ingress/status.go

```diff
@@ -979,6 +988,16 @@ func updateIngressControllerAWSEIPAllocationStatus(ic *operatorv1.IngressControl
 	}
 }
 
+// updateIngressControllerFloatingIPOpenStackStatus mutates the provided IngressController object to
+// sync its status to the effective floatingIP on the LoadBalancer-type service.
+func updateIngressControllerFloatingIPOpenStackStatus(ic *operatorv1.IngressController, service *corev1.Service) {
+	// ProviderParameters should be initialized by setDefaultProviderParameters
+	// when an IngressController is admitted, so we don't need to initialize here.
+	if fip := getLoadBalancerIPFromService(service); fip != "" {
+		ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.OpenStack.FloatingIP = fip
+	}
```

Comment:
Now that I think about it more, I think we should have a NIL check here as we do with `updateIngressControllerAWSEIPAllocationStatus` and `updateIngressControllerAWSSubnetStatus`. `setDefaultProviderParameters` does initialize `ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.OpenStack`, but that only happens on first admission, and a user can remove it later via `oc apply`, then this would be a nil pointer exception. We are trying to fix those in the code.

Additionally, I think you don't really need to check if `fip != ""` because string `"" ` and `nil` are the same thing since it's not a pointer.

```suggestion
if ic.Status.EndpointPublishingStrategy != nil &&
		ic.Status.EndpointPublishingStrategy.LoadBalancer != nil &&
		ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters != nil &&
		ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.OpenStack != nil {
		ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.OpenStack.FloatingIP = getLoadBalancerIPFromService(service)
}
```